### PR TITLE
Reproduction

### PR DESCRIPTION
**Pull request by graberyanis**
- reproduction:
- function that will take 2 networks as inputs and creates a new one from them, copying weights and biases from the 2 "parent" networks, from which one they are taken is randomized.
- the structure of the neural network will stay the same for every organism and every generation for this function to work proerly
- simulation:
- normalization function to get all lifetime values to add up to 1
- selection function to pick a random organism as parent, while giving ones with higher lifetimes a better chance
- new_gen function to create a new equally big population by creating new organisms each with their own network, combined from 2 parents from the last generation

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -20,9 +20,13 @@ def __init__(self):
...
         self.food_spread_chance = 5 # final chance wile be 1 / spread_chance
 
-        self.population_size = 500
+        self.population_size = 50
 
-        self.organism_min_life = 60
-        self.organism_max_life = 100
+        self.organism_eyes_size = 2
+
+        self.nn_layer_sizes = ((self.organism_eyes_size * 2 + 1) ** 2, 5, 9)
```
Comment (webermael) [inline src/config.py:27]: Fancy

File: src/simulation.py [hunk 2]
```diff
@@ -31,13 +36,53 @@ def run(self):
...
+        new_population = Population(self.config)
+        self.normalize_lifetime()
+
+        for organism in range(self.config.population_size):
+            new_org = Organism(self.config, 
+                               new_population.organisms[organism].position.x, 
+                               new_population.organisms[organism].position.y, 
+                               reproduction(self.config, self.selection(self.population.organisms), self.selection(self.population.organisms)))
+            new_population.organisms[organism] = new_org
+        return new_population
```
Comment (webermael) [inline src/simulation.py:75]: Very clean code, I think our code structure works.

File: src/simulation.py [hunk 3]
```diff
@@ -31,13 +36,53 @@ def run(self):
...
+
     def reset(self):
         self.generation_duration = self.config.generation_duration
         self.world = World(self.config)
-        self.population = Population(self.config)
+        self.population = self.new_gen()
         self.screen = screen_init(self.config)
     
     def evolve(self):
+        self.running = True
```
Comment (webermael) [inline src/simulation.py:85]: Thx

File: src/config.py [hunk 4]
```diff
@@ -20,9 +20,13 @@ def __init__(self):
...
+
+        self.organism_min_life = 200
+        self.organism_max_life = 400
```
Comment (webermael) [inline src/config.py:30]: Will the lifetime of the organism stay randomized?

**Review by webermael — APPROVED** [review]
The project is taking shape!

File: src/config.py (hunk 4 above)
Comment (graberyanis) [reply src/config.py:30]: Was planning to change it on a *different* branch ;)

Comment (webermael) [reply src/config.py:30]: Got me there :-D

**Pull request merged**